Downstream Analyses using the R package SCEPTRE
-----------------------------------------------

The assignment outputs obtained by crispat can serve as direct input for differential expression tests. One option is to use the R package **SCEPTRE** which is tailored for differential expression testing in single-cell CRISPR screens and also calculates the number of significant target genes (and their log2 fold change), as well as the number of false discoveries (for non-targeting gRNAs). More details on the SCEPTRE pipeline and all its options can be found [here](https://timothy-barry.github.io/sceptre-book/). In the file `SCEPTRE_discovery.R`, we provide an example of how the assignments obtained by crispat can be used as input for the SCEPTRE pipeline using some helper functions defined in `SCEPTRE_utils.R`. The `SCEPTRE_discovery.R` script is executed using the `run_discovery_analysis.sh` in which the assignment file and the gRNA-target data frame file paths are specified. All code used for the analyses shown in our paper such as the comparison of the SCEPTRE output across methods can be found in a separate [github repository](https://github.com/velten-group/crispat_analysis).
